lib/lightning/projects.ex

```diff
@@ -286,22 +287,40 @@ defmodule Lightning.Projects do
       {:error, %Ecto.Changeset{}}
 
   """
-  def update_project(%Project{} = project, attrs) do
+  def update_project(%Project{} = project, attrs, user \\ nil) do
     changeset = Project.changeset(project, attrs)
 
-    case Repo.update(changeset) do
-      {:ok, updated_project} ->
+    Multi.new()
+    |> Multi.update(:project, changeset)
+    |> maybe_audit_changes(changeset, user)
```

Comment:
Here if you want a whole pack of multi for audit you can use a Multi.merge. It would adhere to the practice of receiving the previous changes (multi operations) via higher order function param. It provides a built-in way similar to but mapped to the multi name (e.g `fn %{project: project} ->` to get the project):
```
  defp filter_changes(%{changes: changes} = changeset, field) do
    changeset |> Map.merge(%{changes: changes |> Map.take([field])})
  end
```